osclib/accept_command.py

```diff
@@ -128,20 +179,42 @@ def cleanup(self, project):
 
     def accept_other_new(self):
         changed = False
+
+        if self.api.delreq_review:
+            rqlist = self.find_virtually_accepted_requests(self.api.project)
+            for req in rqlist:
+                if self.virtual_accept_request_has_no_binary(self.api.project, req['packages'][0]):
+                    # Accepting delreq-review review
+                    self.api.do_change_review_state(req['id'], 'accepted',
+                                                    by_group=self.api.delreq_review,
+                                                    message='Virtually accepted delete {}'.format(req['packages'][0]))
+
         rqlist = self.find_new_requests(self.api.project)
         if self.api.cnonfree:
             rqlist += self.find_new_requests(self.api.cnonfree)
 
         for req in rqlist:
             oldspecs = self.api.get_filelist_for_package(pkgname=req['packages'][0], project=self.api.project, extension='spec')
             print 'Accepting request %d: %s' % (req['id'], ','.join(req['packages']))
+            if req['type'] == 'delete':
+                # Remove devel project/package tag before accepting the request
+                self.remove_obsoleted_develtag(self.api.project, req['packages'][0])
             change_request_state(self.api.apiurl, str(req['id']), 'accepted', message='Accept to %s' % self.api.project)
             # Check if all .spec files of the package we just accepted has a package container to build
             self.create_new_links(self.api.project, req['packages'][0], oldspecs)
             changed = True
 
         return changed
 
+    def remove_obsoleted_develtag(self, project, package):
+        xpath = {
+            'package': "@project='%s' and devel/@project='%s' and devel/@package='%s'" % (project, project, package),
+        }
+        collection = search(self.api.apiurl, **xpath)['package']
+        pkglist = [p.attrib['name'] for p in collection.findall('package')]
+        for pkg in pkglist:
```

Comment:
Any reason to build a list and then iterate over it rather than just looping over the original data structure?

```python
for pkg in collection.findall('package'):
    set_devel_project(self.api.apiurl, project, pkg.attrib['name'], devprj=None)
```

Reply:
just different coding style not you like...

Reply:
Indeed, nothing incorrect about it, just curious as typically looping twice is worse than looping once.

Reply:
Well, loops an extra time and allocates an entire extra array. Again time, but I see no upside.